source/Symbol/SwiftASTContext.cpp

```diff
@@ -1335,22 +1335,23 @@ static bool DeserializeCompilerFlags(swift::CompilerInvocation &invocation,
   return false;
 }
 
-static void printASTValidationInfo(
+static void printASTValidationError(
     llvm::raw_ostream &errs,
     const swift::serialization::ValidationInfo &ast_info,
     const swift::serialization::ExtendedValidationInfo &ext_ast_info,
-    const Module &module, StringRef module_buf, bool invalid_name,
+    Module &module, StringRef module_buf, bool invalid_name,
     bool invalid_size) {
   const char *error = getImportFailureString(ast_info.status);
-  errs << error;
-  if (invalid_name) {
-    error = "The module has an invalid name.";
-    errs << ' ' << error;
-  }
-  if (invalid_size) {
-    error = "The module has an invalid file size.";
-    errs << ' ' << error;
-  }
+  errs << "AST validation error";
+  if (!invalid_name)
+    errs << " in \"" << ast_info.name << '"';
+  errs << ": ";
+  if (invalid_size)
+    errs << "The serialized module is corrupted.";
+  else if (invalid_name)
+    errs << "The serialized module has an invalid name.";
+  else
```

Comment:
That's intentional, if the size or the name is invalid, we don't even parse, and the error will be, "The module file is malformed in some way." and since we know the "some way" we explicitly be more specific. I'll add a comment.